sound/soc/sof/intel/hda-loader.c

```diff
@@ -372,3 +372,148 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	dev_err(sdev->dev, "error: load fw failed err: %d\n", ret);
 	return ret;
 }
+
+// skl/kbl enable core and code loader DMA has some difference with apl/cnl
+// add the APIs for skl/kbl
+static int cl_stream_prepare_skl(struct snd_sof_dev *sdev, unsigned int format,
+				 unsigned int size, struct snd_dma_buffer *dmab,
+				 int direction)
+{
+	u32 tag = 0;
+	return tag;
+}
+
+static int cl_dsp_init_skl(struct snd_sof_dev *sdev, const void *fwdata,
+			   u32 fwsize)
+{
+	int tag, ret, i;
+	u32 hipcie;
+	u32 reg;
+	const struct sof_intel_dsp_desc *chip = sdev->hda->desc;
+
+	if (hda_dsp_core_is_enabled(sdev, HDA_DSP_CORE_MASK(0))) {
+		ret = hda_dsp_core_stall_reset(sdev, HDA_DSP_CORE_MASK(0));
+		if (ret < 0)
+			goto err;
+
+		ret = hda_dsp_core_run(sdev, HDA_DSP_CORE_MASK(0));
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: dsp core start failed %d\n",
+				ret);
+			ret = -EIO;
+			goto err;
+		}
+	} else {
+		ret = hda_dsp_core_reset_power_down(sdev, HDA_DSP_CORE_MASK(0));
+		if (ret < 0) {
+			dev_err(sdev->dev, "dsp core0 disable fail: %d\n", ret);
+			return ret;
+		}
+		ret = hda_dsp_enable_core(sdev, HDA_DSP_CORE_MASK(0));
+	}
+
+	/* prepare DMA for code loader stream */
+	tag = cl_stream_prepare_skl(sdev, 0x40, fwsize, &sdev->dmab,
```

Comment:
Just set tag = 0 at start of this function if it's always going to be 0 ???

Reply:
Like answered above, the next patch will get the true tag value, this is only for debug.